index.js

```diff
@@ -296,21 +312,64 @@ function createFocusTrap(element, userOptions) {
   // kind of need to capture the action at the keydown phase.
   function checkTab(e) {
     updateTabbableNodes();
-    if (e.shiftKey && e.target === state.firstTabbableNode) {
-      e.preventDefault();
-      tryFocus(state.lastTabbableNode);
-      return;
+    const isShift = e.shiftKey;
+
+    if (isShift) {
+      const startOfGroupIndex = state.tabbableGroups.findIndex(
+        ({ firstTabbableNode }) => e.target === firstTabbableNode
+      );
+
+      const isStartOfAGroup = startOfGroupIndex !== -1;
+      if (isStartOfAGroup) {
+        e.preventDefault();
+
+        const isFirstGroup = startOfGroupIndex === 0;
+        const lastGroupIndex = state.tabbableGroups.length - 1;
+        const previousGroupIndex = startOfGroupIndex - 1;
+        const destinationGroupIndex = isFirstGroup
+          ? lastGroupIndex
+          : previousGroupIndex;
+
+        const destinationGroup = state.tabbableGroups[destinationGroupIndex];
+        const destinationNode =
+          destinationGroup && destinationGroup.lastTabbableNode;
+
+        tryFocus(destinationNode);
+
+        return;
+      }
     }
-    if (!e.shiftKey && e.target === state.lastTabbableNode) {
-      e.preventDefault();
-      tryFocus(state.firstTabbableNode);
-      return;
+    if (!isShift) {
```

Comment:
Can this be an `else` statement, no need for `return` statements on 339 and 365...?

Reply:
Done.